### DISCUSSION TITLE
hl-todo not highlighting in line-comments in scala-ts-mode

### DISCUSSION THREAD

**Discussion by ravenjoad** [tarsius/hl-todo#83] [Support] [answered]
I am testing out [scala-ts-mode](https://github.com/KaranAhlawat/scala-ts-mode), and it is working nearly identically to the normal scala-mode, except for one thing I cannot seem to figure out.

In scala-ts-mode, hl-todo does not highlight any of the keywords in `hl-todo-keyword-faces` in **line** comments. But the highlighting works in **block** comments.

In the image, scala-tree-sitter is shown, with the tree-sitter explorer for that file. On the right is the same file in scala-mode and is what I expect.

![image](https://github.com/user-attachments/assets/3dd27f7a-8ae3-4a8f-80d9-5cc5a1d40740)

I have turned off the LSP and disabling flymake, and nothing changed.

**Comment by tarsius** [comment, marked as answer]
From the lower window and quickly peaking at the code, I get the impression, that the two kinds of comments are implemented quite differently.

In both places, check if `hl-todo--inside-comment-or-string-p` (i.e., `(nth 8 (syntax-ppss))`) returns non-nil. Hl-todo only highlights if that is so.

**Reply by ravenjoad** [reply]
You're right. When point is in the sequence of line-comments, `M-: (hl-todo--inside-comment-or-string-p) <RET>` returns `nil`. When in the sequence of block comments, I get an integer back.

**Reply by ravenjoad** [reply]
Oh. It even has an open issue about this. https://github.com/KaranAhlawat/scala-ts-mode/issues/17

I changed `scala-ts--syntax-table`'s comment entries and hl-todo automatically worked. I'll submit that there.

Thanks for the help!
